src/senselab/audio/tasks/text_to_speech/api.py

```diff
@@ -1,37 +1,109 @@
-        model (SenselabModel): The model used for synthesis
-            (Default is "suno/bark").
+        target (Optional[List[Tuple[Audio, Optional[str]]]]):
+            A list where each element is a tuple of target audio and optional transcript.
+        model (HFModel): The model used for synthesis (Default is "suno/bark").
         language (Optional[Language]): The language of the text (default is None).
         device (Optional[DeviceType]): The device to run the model on (default is None).
+        **kwargs: Additional keyword arguments to pass to the synthesis function.
+            Depending on the model used (e.g., Hugging Face or Mars5-TTS), additional arguments
+            may be required. You can find details in the documentation of each function
+            (e.g., `synthesize_texts_with_transformers` or `synthesize_texts_with_mars5tts`)
+            and in the card of each model.
 
     Returns:
         List[Audio]: The list of synthesized audio objects.
-
-    Todo:
-        - Include more models
-        - Include voice cloning for TTS
     """
+    target_audios: List[Audio] = []
+    target_transcripts: List[Optional[str]] = []
+    if target is not None:
+        target_audios = []
+        target_transcripts = []
+
+        for t in target:
+            if isinstance(t, tuple):
+                audio, transcript = t
+                if isinstance(audio, Audio):
+                    target_audios.append(audio)
+                else:
+                    raise TypeError("Expected the first element of the tuple to be an Audio object.")
+
+                if isinstance(transcript, str):
+                    target_transcripts.append(transcript)
+                else:
+                    raise TypeError("Expected the second element of the tuple to be a string.")
+            elif isinstance(t, Audio):
+                target_audios.append(t)
+                target_transcripts.append(None)
+            else:
+                raise TypeError("Expected elements of target to be either Audio objects or tuples of (Audio, str).")
+
+        if len(texts) != len(target_audios):
+            raise ValueError("The lengths of texts and target audios must be the same.")
+        if any(t is not None for t in target_transcripts) and len(texts) != len(target_transcripts):
+            raise ValueError("The lengths of texts and target transcripts must be the same.")
+
     if isinstance(model, HFModel):
-        return HuggingFaceTTS.synthesize_texts_with_transformers(texts=texts, model=model, device=device)
+        return HuggingFaceTTS.synthesize_texts_with_transformers(texts=texts, model=model, device=device, **kwargs)
+    elif isinstance(model, TorchModel):
+        if model.path_or_uri == "Camb-ai/mars5-tts":
+            # Converting target to the required format for Mars5TTS
+            target_for_mars5tts = _get_audio_and_transcript_targets(target_audios, target_transcripts)
```

Comment:
Why do you unzip `target` and then rezip `target` instead of just using `target` directly here?

Reply:
Could you just do the checks without creating new variables / functions to unzip and rezip? Like have the loop in the beginning just do the checks, and if they pass, send `target` into `mars5tts`?

Reply:
the problem with that approach is that you cannot guarantee data type consistency and mypy will complain:
`Argument "target" to "synthesize_texts_with_mars5tts" of "Mars5TTS" has incompatible type "list[tuple[Audio, str | None]] | None"; expected "list[tuple[Audio, str]]"  [arg-type]`

Reply:
I can take a look at this later. there are definitely approaches that will make mypy happy that don't require zipping/unzipping